inkstitch.py

```diff
@@ -31,55 +98,95 @@
 from lib.i18n import _
 from lib.utils import restore_stderr, save_stderr
 
-# ignore warnings in releases
-if getattr(sys, 'frozen', None):
+# file DEBUG exists next to inkstitch.py - enabling debug mode depends on value of debug_type in DEBUG file
+if debug_type != 'none':
+    debug.enable(debug_type)
+    # check if debugger is really activated
+    debug_active = bool((gettrace := getattr(sys, 'gettrace')) and gettrace())
+
+# ignore warnings in releases - see warnings.warn()
+if running_as_frozen or not debug_active:
     import warnings
     warnings.filterwarnings('ignore')
 
-logger = logging.getLogger('shapely.geos')
+# set logger for shapely
+logger = logging.getLogger('shapely.geos')  # attach logger of shapely, from ver 2.0.0 all logs are exceptions
 logger.setLevel(logging.DEBUG)
-shapely_errors = StringIO()
+shapely_errors = StringIO()                # in memory file to store shapely errors
 ch = logging.StreamHandler(shapely_errors)
 ch.setLevel(logging.DEBUG)
 formatter = logging.Formatter('%(name)s - %(levelname)s - %(message)s')
 ch.setFormatter(formatter)
 logger.addHandler(ch)
 
+# pop '--extension' from arguments and generate extension class name from extension name
 parser = ArgumentParser()
 parser.add_argument("--extension")
 my_args, remaining_args = parser.parse_known_args()
 
-if os.path.exists(os.path.join(os.path.dirname(os.path.realpath(__file__)), "DEBUG")):
-    debug.enable()
-
-profiler = None
-if os.path.exists(os.path.join(os.path.dirname(os.path.realpath(__file__)), "PROFILE")):
-    profiler = cProfile.Profile()
-    profiler.enable()
-
 extension_name = my_args.extension
 
 # example: foo_bar_baz -> FooBarBaz
 extension_class_name = extension_name.title().replace("_", "")
 
 extension_class = getattr(extensions, extension_class_name)
-extension = extension_class()
+extension = extension_class()  # create instance of extension class - call __init__ method
+
+# extension run(), but we differentiate between debug and normal mode
+# - in debug or profile mode we run extension or profile extension
+# - in normal mode we run extension in try/except block to catch all exceptions and hide GTK spam
+if debug_active or profile_type != "none":  # if debug or profile mode
+    print(f"Extension:'{extension_name}' Debug active:{debug_active} type:'{debug_type}' "
```

Comment:
I miiight end up hating this, we'll see.  I keep my DEBUG file set most of the time.  If I don't have debug mode active in PyCharm, then it just continues on, but I still get to see warnings.  Not sure if it might be annoying to have this message every time I run an extension.

Reply:
Message is just for a better initial acquaintance with the code; it disappears.

But the second part concerns me. I assume that if I have an active DEBUG file, the script switches to a waiting mode and waits for the debugger to connect and doesn't continue without the debugger.

In VSCode, I haven't found a way for the script to check if there is a debugger waiting somewhere that it could connect to. That's why, for VSCode, the script stops and waits for a connection. Perhaps PyCharm has a default port on which the debugger can communicate and check if a debugger is waiting on that port.
It's inconvenient when launched directly from Inkscape because it gets stuck until the debugger is activated.

Reply:
> Message is just for a better initial acquaintance with the code; it disappears.

It doesn't disappear though.  Every time we run an Ink/Stitch extension with debug enabled, Inkscape will say "the extension produced output" and show this message, and the dev will have to dismiss the dialog.

> But the second part concerns me. I assume that if I have an active DEBUG file, the script switches to a waiting mode and waits for the debugger to connect and doesn't continue without the debugger.
> 
> In VSCode, I haven't found a way for the script to check if there is a debugger waiting somewhere that it could connect to. That's why, for VSCode, the script stops and waits for a connection. Perhaps PyCharm has a default port on which the debugger can communicate and check if a debugger is waiting on that port.
It's inconvenient when launched directly from Inkscape because it gets stuck until the debugger is activated.

Ah, yes, PyCharm (and PyDev) work the other way around: the IDE opens a listening port and waits, and the script connects when it wants to be debugged.  The pydev/pycharm code in debug.py catches and ignores connection errors.  If PyCharm isn't waiting in debug mode, Ink/Stitch doesn't wait to be debug, it just continues on as normal.